### PR TITLE
Boiler Smoke Upgrade: Acid smoke damages barricades/tanks. Neuro transfers +1U reagent

### PR DESCRIPTION
**Pull request by Surrealaser**
:cl: by Surrealistik
add: Boiler acid smoke deals damage to tanks at a rate of roughly 30 per second of exposure, and most barricades at 2-3 damage per second of exposure. Subject to tweaks.
tweak: Neuro gas transfers +1U neurotoxin to offset the higher neuro meta rate.
/:cl:

### CODE REVIEW

File: code/game/objects/structures/barricade.dm [hunk 1]
```diff
@@ -933,3 +933,15 @@
...
+	health -= 2
+	update_health()
+
+/obj/structure/barricade/wooden/acid_smoke_damage(var/obj/effect/particle_effect/smoke/S) //wood isn't as reactive with beno acid. Also for balance reasons because wood sucks total ass.
+	health -= 2
+	update_health()
+
+/obj/structure/barricade/snow/acid_smoke_damage(var/obj/effect/particle_effect/smoke/S) //snow isn't as reactive with acid. Also for balance reasons because snow sucks total ass.
+	health -= 2
+	return
```
Comment (DominikPanic) [inline code/game/objects/structures/barricade.dm]: Unnecessary return.

File: code/game/objects/effects/effect_system/smoke.dm [hunk 2]
```diff
@@ -280,16 +280,27 @@
 	spread_speed = 7
 	amount = 1 //Amount depends on Boiler upgrade!
 
+/obj/effect/particle_effect/smoke/xeno_burn/process()
+    ..()
```
Comment (DominikPanic) [inline code/game/objects/effects/effect_system/smoke.dm]:
```suggestion
    . = ..()
```

**Comment by Monikawales** [comment]
The best tweak to prevent 4 hour turtle-fob sieges

File: code/game/objects/structures/barricade.dm (hunk 1 above)
Comment (Surrealaser) [reply code/game/objects/structures/barricade.dm]: Looks like copypasta gone wrong

File: code/game/objects/effects/effect_system/smoke.dm [hunk 3]
```diff
@@ -280,16 +280,27 @@
...
+/obj/effect/particle_effect/smoke/xeno_burn/process()
+    . = ..()
+    var/turf/T = get_turf(src)
+    if(!T)
+        qdel(src)
+    apply_smoke_effect(T)
+
 /obj/effect/particle_effect/smoke/xeno_burn/apply_smoke_effect(turf/T)
 	for(var/mob/living/L in T)
+		//to_chat(world, "DEBUG xeno_burn/apply_smoke_effect: Turf: [T] Living: [L] ")
```
Comment (spookydonut) [inline code/game/objects/effects/effect_system/smoke.dm]: either remove these comments or put them inside ifdefs
```
#ifdef BOILERSMOKEDEBUG
to_chat(world etc
#endif```

File: code/game/objects/effects/effect_system/smoke.dm [hunk 4]
```diff
@@ -280,16 +280,27 @@
 	spread_speed = 7
 	amount = 1 //Amount depends on Boiler upgrade!
 
+/obj/effect/particle_effect/smoke/xeno_burn/process()
+    . = ..()
+    var/turf/T = get_turf(src)
```
Comment (Rohesie) [inline code/game/objects/effects/effect_system/smoke.dm]: Why do you need to use `get_turf` instead of `loc` here?

**Comment by DominikPanic** [comment]
Do you still plan to work on this?

**Comment by Surrealaser** [comment]
Didn't Ghommie effectively do this with his refactor?

If so, I just need to make a separate PR for the Defiler.

**Comment by DominikPanic** [comment]
Alright

**Pull request closed without merging**